examples/systemcalls/systemcalls.c

```diff
@@ -59,6 +74,66 @@ bool do_exec(int count, ...)
  *
 */
 
+    int status_int;
+
+    // PID to store the child PID
+    pid_t child;   
+
+    // Checking if the command path is absolute
+    if(command[0][0] != '/')
+    {
+        printf("\nError: The file path entered is '%s' and is not absolute", command[0]);
+        return false;
+    }
+
+    // Stores the return in the child PID variable
+    child = fork();
+
+    if(-1 == child)
+    {
+        // Error handling the fork()
+        printf("\nError: fork() returned error. Error code: %d", errno);
+        return false;
+    }
+    else if (0 == child)
+    {
+        // Child executes this block
+        status_int = execv(command[0], command);
+
+        if (-1 == status_int)
+        {
+            printf("\nError: execv() returned error. Error code: %d", errno);
+            return false;
+        }
+    }
+    else
+    {
+        // Parent executes this block
+        pid_t status_pid = waitpid(child, &status_int, 0);
+
+        if (-1 == status_pid)
+        {
+            printf("\nError: waitpid() returned error. Error code: %d", errno);
+            return false;
+        }
+        // Checking id exited normally
+        else if(WIFEXITED(status_int))
+        {
+            // Evaluates to the lower order of 8 bits of status if the child is passed though exit() or main()
+            if(0 != WEXITSTATUS(status_int))
+            {
+                return false;
```

Comment:
Missing cleanup in these code paths.
Also see discussion in class about avoiding duplication between these two very similar functions.